### PR TITLE
Some Coverity fixes

### PR DESCRIPTION
**Pull request by jsquyres**
I did these while otherwise waiting for other things this morning.

Most are trivial / easy, but I want to get some feedback on some of them before bringing them to master.  I'll ping individual code owners to get their feedback on relevant commits.

### CODE REVIEW

File: orte/tools/orterun/orterun.c [hunk 1]
```diff
@@ -2266,20 +2265,8 @@ static int process(char *orig_line, char *basename, opal_cmd_line_t *cmd_line,
...
-            free(tmp);
-            tmp = opal_argv_join(orterun_argv, ' ');
-            orte_show_help("help-orterun.txt", "debugger requires -np",
-                           true, (*new_argv)[0], argv[0], tmp,
-                           (*new_argv)[0]);
-            /* Fall through to free / fail, below */
-        } 
-
         /* Some debuggers do not support launching MPMD */
-        else if (single_app && NULL != strchr(tmp, ':')) {
```
Comment (jsquyres) [inline orte/tools/orterun/orterun.c]: @rhc54 This looked like totally dead code (and therefore also enabled me to remove a block from help-orterun.txt).

Can you confirm that it's ok to remove this block?

File: ompi/mca/bcol/basesmuma/bcol_basesmuma_allreduce.c [hunk 2]
```diff
@@ -268,7 +269,7 @@ int bcol_basesmuma_allreduce_intra_fanin_fanout(bcol_function_args_t *input_args
 {
     /* local variables */
     mca_bcol_basesmuma_module_t *bcol_module = (mca_bcol_basesmuma_module_t *) c_input_args->bcol_module;
-    int buff_idx = buff_idx = input_args->src_desc->buffer_index;
```
Comment (jsquyres) [inline ompi/mca/bcol/basesmuma/bcol_basesmuma_allreduce.c]: @shamisp @manjugv There was no deep/secret meaning for assigning `buff_idx = buff_idx = some_other_value`, right?

**Comment by mellanox-github** [comment]
* Coverity found 7 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_279/c/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/279/
Test PASSed.

File: orte/tools/orterun/orterun.c (hunk 1 above)
Comment (rhc54) [inline orte/tools/orterun/orterun.c]: I don't think that is correct, Jeff. The correct change is to add "used_num_procs=true;" right after line 2225

Comment (jsquyres) [inline orte/tools/orterun/orterun.c]: Got it; I'll update.  Thanks.

**Comment by mellanox-github** [comment]
* Coverity found 7 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_280/c/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/280/
Test PASSed.

**Comment by mellanox-github** [comment]
* Coverity found 7 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_281/c/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/281/
Test PASSed.

**Comment by mellanox-github** [comment]
* Coverity found 7 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-master-pr//ws/cov_build/oshmem_282/c/output/errors/index.html <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/282/
Test PASSed.

**Pull request merged**